### DISCUSSION TITLE
添加外部链接能否增加对localhost的支持，直接指向服务器

### DISCUSSION THREAD

**Discussion by sesame-foxes** [IceWhaleTech/CasaOS#648] [Ideas] [answered]
添加外部链接时需要输入完整ip，是否可以加入对localhost或者127.0.0.1的支持，让url可以直接指向当前服务器。

**Comment by zhanghengxin** [comment]
但是这样子，你在其他设备上访问，就会指向本地啊

A jump error occurs when other devices access it as your saying

**Comment by zhanghengxin** [comment]
嗨，大佬，有兴趣加入微信群嘛？详细的沟通一下你想要解决的问题！

Hi, Guy, interested in joining a wechat group? Communicate the problem you want to solve in detail!  @sesame-foxes

**Comment by zhanghengxin** [comment]
我的提议是自动添加协议头部，避免重复手动输入，对于localhost 我始终认为这并不应该支持，因为这在任何其他设备无法访问。

my proposal is add a default protocol to avoid manual typing, and i think "localhost" is meaningful, -- it can't running in other devices 

@sesame-foxes  looking forward and your reply

**Comment by sesame-foxes** [comment, marked as answer]
> 我的提议是自动添加协议头部，避免重复手动输入，对于localhost 我始终认为这并不应该支持，因为这在任何其他设备无法访问。
> 
> my proposal is add a default protocol to avoid manual typing, and i think "localhost" is meaningful, -- it can't running in other devices
> 
> @sesame-foxes looking forward and your reply

抱歉，前几天由于网络问题，无法直连github。我的描述可能并不准确。当我在设备上安装casaos之后，自行添加的本地（本设备）的应用，需要输入完整ip。我的设想是有localhost之类的参数，可以直接指向该设备本身。这样添加的外部应用，在以后设备ip变换之后依然可以直接访问而不需要修改链接ip。
比方说使用local:port这种写法，或直接 :port 就可以直接指向该设备ip

**Comment by zhanghengxin** [comment]
unbelievable！！！ 但这个不是很常规的操作，我需要转给产品同学，让他形成一个广为人知，容易被发现的功能。

i need to turn it over to PM， and ask him to think about how to do it so that it's easy to find and using. 

@ETWang1991

**Comment by sesame-foxes** [comment]
> unbelievable！！！ 但这个不是很常规的操作，我需要转给产品同学，让他形成一个广为人知，容易被发现的功能。
> 
> i need to turn it over to PM， and ask him to think about how to do it so that it's easy to find and using.
> 
> @ETWang1991

我的一点想法：
window.open('http://'+document.domain+':'+port,'_blank');

**Comment by ETWang1991** [comment]
window.open('http://'+document.domain+':'+port,'_blank');
I have documented this idea and will remove the write restriction first in the latest version.
关于这个想法我已记录，最近版本会先解除写入的限制
